internal/cli/search/index_opts.go

```diff
@@ -170,14 +165,59 @@ func (opts *IndexOpts) NewSearchIndex() (*atlasv2.SearchIndexCreateRequest, erro
 	return i, nil
 }
 
+func (opts *IndexOpts) readCreateFromFile() (*atlasv2.SearchIndexCreateRequest, error) {
+	index := &atlasv2.SearchIndexCreateRequest{}
+	if err := opts.loadStrict(index); err != nil {
+		return nil, err
+	}
+	if err := file.Load(opts.Fs, opts.Filename, index); err != nil {
+		return nil, fmt.Errorf(failedToLoadIndexMessage, err.Error())
+	}
+
+	if index.Type == nil {
+		index.Type = pointer.Get(DefaultType)
+	}
+
+	return index, nil
+}
+
+func (opts *IndexOpts) loadStrict(v interface{}) error {
+	file, err := opts.Fs.Open(opts.Filename)
+	if err != nil {
+		return fmt.Errorf("could not open file: %w", err)
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(file)
+	if err != nil {
+		return fmt.Errorf("could not read file: %w", err)
+	}
+	decoder := json.NewDecoder(bytes.NewReader(data))
+	decoder.DisallowUnknownFields()
+
+	if err := decoder.Decode(v); err != nil {
+		_, _ = log.Warningf("See expected index format for creates and updates at https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Atlas-Search/.\n")
+		return fmt.Errorf("could not decode JSON: %w", err)
+	}
+
+	return nil
```

Comment:
You could use `binary.Read` to get the `&atlasv2.SearchIndexCreateRequest{}` from the `data` binary that you have here so that you don't have to reopen the same file in `readCreateFromFile`